x-pack/plugins/endpoint/common/types.ts

```diff
@@ -168,39 +169,44 @@ export type AlertEvent = Immutable<{
     module: string;
     type: string;
   };
+  endpoint: {
+    policy: {
+      id: string;
+    };
+  };
   process: {
     code_signature: {
       subject_name: string;
       trusted: boolean;
     };
-    command_line: string;
-    domain: string;
+    command_line?: string;
+    domain?: string;
     pid: number;
-    ppid: number;
+    ppid?: number;
     entity_id: string;
-    parent: {
+    parent?: {
       pid: number;
       entity_id: string;
     };
     name: string;
     hash: HashFields;
-    pe: {
+    pe?: {
       imphash: string;
     };
-    executable: string;
-    sid: string;
+    executable?: string;
+    sid?: string;
     start: number;
-    malware_classifier: MalwareClassifierFields;
+    malware_classifier?: MalwareClassifierFields;
```

Comment:
just a general question, what was the criteria of making some of these fields optional?

Reply:
I made as many things optional as I could without breaking the type checker